pkg/controller/service/servicecontroller.go

```diff
@@ -339,14 +339,23 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-
-		err := s.createLoadBalancer(service, namespacedName)
+		_, exists, err := s.balancer.GetLoadBalancer(s.loadBalancerName(service), s.zone.Region)
```

Comment:
So IIUC the pool existed and had the right nodes so we failed because the api returned StatusConflict, and kept retrying. What if the pool existed but had the wrong nodes? 

It sounds like ideally, if the pool exists (StatusConflict) we should check that it has the right nodes. And if it does, `createLoadBalancer` should just log the StatusConflict and not return an error. If it has the wrong nodes, `createLoadBalancer` should fix this. If it can't fix this, it should return *that* error.

Thoughts?